src/parsing/parser.ts

```diff
@@ -354,14 +358,15 @@ export function parseForeignCallDefinition(
 
   const returnType: number = PType.indexOf(syntax.returnType);
 
-  const parameterTypes: number[] = splitFunctionInput(syntax.function).map(
-    (param) => PType.indexOf(param)
-  );
-
-  var valuesToPass: number[] = syntax.valuesToPass
-    .split(",")
-    .filter((input: string) => !isNaN(Number(input)))
-    .map((input: string) => Number(input));
+  var parameterTypes: number[] = [];
```

Comment:
`parameterTypes` assignment can use filter and map
```
const parameterTypes = splitFunctionInput(syntax.function).map(args => PT.find(pt => arg == pt.name)).map(pt => PT.find(pt.enumeration).filter(pt => pt).map(pt => pt.enumeration)
```
although since pt.enumeration is the same as the index of the object we can also use
```
const parameterTypes = splitFunctionInput(syntax.function).map(args => PT.findIndex(pt => arg == pt.name))
```

Reply:
Agreed, a convenience function that accepts a function name and then looks it up in the PT array and rreturns its enumeration might be cleanest